Video Media Bundle

vijaynandwani opened the pull request:
Here is a PR for video media bundle. The module video_embed_field comes with a lot of tests already. Here is a sample code from file video_embed_field/modules/video_embed_media/tests/src/Functional/BundleTest.php

> // Create a video, populating both video URL fields.
>     $this->drupalGet('media/add/video_bundle');
>     $this->submitForm([
>       'name[0][value]' => 'Another Video!',
>       'field_media_video_embed_field[0][value]' => 'https://www.youtube.com/watch?v=XgYu7-DQjDQ',
>       'field_new_video_field[0][value]' => 'https://www.youtube.com/watch?v=gnERPdAiuSo',
>     ], t('Save'));
> 
> ```
> // We should see the newly configured video thumbnail, but not the original.
> $this->assertContains('video_thumbnails/gnERPdAiuSo.jpg', $this->getSession()->getPage()->getHtml());
> $this->assertNotContains('video_thumbnails/XgYu7-DQjDQ.jpg', $this->getSession()->getPage()->getHtml());
> ```

It checks if the video is being added with the correct thumbnail or not.
The modules comes with a lot of other tests already. Let me know if there are still some tests you want me to work on.

vijaynandwani committed:
Video Media Bundle

vijaynandwani commented:
There is some problem with the tests. Give me some time to fix it. Please review after that. Thanks :)

paranojik commented on media.info.yml:
```diff
@@ -5,5 +5,8 @@ package: Media
+  - video_embed_media
   - media_entity_image
 
+
```
no need for this extra line here...

vijaynandwani replied:
@paranojik I have mailed you a problem.

vijaynandwani commented:
Okay fixed the tests. It can be reviewed now.

paranojik commented on config/install/core.entity_view_display.media.video.default.yml:
```diff
@@ -0,0 +1,48 @@
+status: true
+dependencies:
+  config:
+    - field.field.media.video.field_media_video_embed_field
```
Let's be consistent here. We named the Image field "field_image". So this should be just "field_video".

vijaynandwani commented:
https://www.drupal.org/node/2743665

slashrsm commented on config/install/media_entity.bundle.video.yml:
```diff
@@ -0,0 +1,14 @@
+new_revision: false
+type_configuration:
+  source_field: field_video
+field_map: {  }
```
Would be nice to map "Video ID" and "Video source human name" to entity fields. We need to add them of course.

vijaynandwani commented:
Added a new commit. 

Enabled the revisions.
Added two new fields
Added field map
Modified the tests to check the field map.

One questions. What is the use of the following thing? Should I remove it?

`
_core:
   default_config_hash: yJ-pzHf0J9QMSSJe1IHY6pjlGdg8J_QG2ex19ipUeyU
`

slashrsm commented on config/install/core.entity_view_display.media.video.default.yml:
```diff
@@ -0,0 +1,60 @@
+    third_party_settings: {  }
+hidden:
+  field_id: true
+  thumbnail: true
```
Hiding this two fields on a form is OK, but we can display them when rendering the entity.

vijaynandwani replied:
@slashrsm the field_id doesn't look nice there. 

Below the video you can see the id of the video. So in case of a youtube video ex: https://www.youtube.com/watch?v=mpbWQbkl8_g the id would be mpbWQbkl8_g and looks weird.

vijaynandwani replied:
And why would you want to see the thumbnail of the video on the page where we are displaying the video?

slashrsm replied:
Should refer to ID only. Any you are right. It will look strange. You can leave it like this.

slashrsm commented on config/install/field.field.media.video.field_id.yml:
```diff
@@ -0,0 +1,18 @@
+field_name: field_id
+entity_type: media
+bundle: video
+label: ID
```
Video ID

slashrsm commented on config/install/field.field.media.video.field_source.yml:
```diff
@@ -0,0 +1,18 @@
+field_name: field_source
+entity_type: media
+bundle: video
+label: Source
```
Video source name

slashrsm commented on src/Tests/VideoBundleTest.php:
```diff
@@ -0,0 +1,56 @@
+        $this->assertEqual($this->testBundle->get('description'), 'Use Video for embedding videos hosted by YouTube, Vimeo, or some other provider.', 'Correct description detected.');
+        $this->assertEqual($this->testBundle->get('type'), 'video_embed_field', 'Correct plugin ID detected.');
+        $this->assertEqual($this->testBundle->get('type_configuration'), $type_configuration, 'Type configuration correct.');
+        $this->assertEqual($this->testBundle->get('field_map'), $field_map, 'Correct field map detected.');
```
Would be nice to have another test that creates entity and checks if this fields were actually populated.

vijaynandwani replied:
Okay would do that.

slashrsm commented on src/Tests/VideoBundleTest.php:
```diff
@@ -0,0 +1,56 @@
+ *
+ * @group media
+ */
+class VideoBundleTest extends WebTestBase {
```
PHPCodeSniffer finds few things in this file: 

```
➜  media git:(1f42b36) phpcs --standard=Drupal --extensions=php,module,inc,install,test,profile,theme,js,css,info,txt,md src/Tests/VideoBundleTest.php    

FILE: ...http/drupal8/docroot/modules/media/src/Tests/VideoBundleTest.php
----------------------------------------------------------------------
FOUND 12 ERRORS AND 2 WARNINGS AFFECTING 13 LINES
----------------------------------------------------------------------
 14 | ERROR   | [x] Line indented incorrectly; expected 2 spaces,
    |         |     found 4
 19 | ERROR   | [x] Line indented incorrectly; expected 2 spaces,
    |         |     found 4
 19 | ERROR   | [ ] If the line declaring an array spans longer than
    |         |     80 characters, each element should be broken into
    |         |     its own line
 21 | ERROR   | [x] Line indented incorrectly; expected 2 spaces,
    |         |     found 4
 26 | ERROR   | [x] Line indented incorrectly; expected 2 spaces,
    |         |     found 4
 28 | ERROR   | [x] Line indented incorrectly; expected 2 spaces,
    |         |     found 4
 31 | ERROR   | [x] Line indented incorrectly; expected 2 spaces,
    |         |     found 4
 32 | ERROR   | [x] Line indented incorrectly; expected 4 spaces,
    |         |     found 8
 33 | ERROR   | [x] Line indented incorrectly; expected 4 spaces,
    |         |     found 8
 40 | ERROR   | [x] Line indented incorrectly; expected 6 spaces,
    |         |     found 8
 41 | WARNING | [x] A comma should follow the last multiline array
    |         |     item. Found: 'field_video'
 46 | WARNING | [x] A comma should follow the last multiline array
    |         |     item. Found: 'field_source'
 55 | ERROR   | [x] Expected 1 blank line after function; 0 found
 56 | ERROR   | [x] The closing brace for the class must have an
    |         |     empty line before it
----------------------------------------------------------------------
PHPCBF CAN FIX THE 13 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 53ms; Memory: 6Mb
```

We should probably check existing files too and fix notices in a separate pull request. CodeSniffer docs: https://www.drupal.org/node/1419988

vijaynandwani replied:
I'll fix these and the ones in the existing files and I'll make sure to pass all the future commits through this.

vijaynandwani commented:
What is the use of the following thing? Should I remove it?

 _core: default_config_hash: yJ-pzHf0J9QMSSJe1IHY6pjlGdg8J_QG2ex19ipUeyU

slashrsm commented:
> One questions. What is the use of the following thing? Should I remove it?

No. From ConfigEntityBase:

> A hash calculated by the config.installer service and added during installation.

vijaynandwani committed:
Fixing files with PHPCS

slashrsm commented on src/Tests/VideoBundleTest.php:
```diff
@@ -0,0 +1,96 @@
+  }
+
+  /**
+   * Tests video media bundle creation from config files.
```
This comment is not true.

slashrsm commented on media.info.yml:
```diff
@@ -6,4 +6,5 @@ core: 8.x
   - media_entity_image
-
+  - video_embed_field
+  - video_embed_media
```
The other day I learned that we support namespacing of dependencies. See how we did it in media entity: https://www.drupal.org/node/2744455

Also see relevant change record: https://www.drupal.org/node/2299747

This might also solve the problem that required  before_script.sh. Not sure, but you should try it out.

vijaynandwani committed:
Fixing dependency namespaces

Modifying travis tests to remove before script
